src/gui/torrentoptionsdialog.cpp

```diff
@@ -241,17 +303,38 @@ TorrentOptionsDialog::TorrentOptionsDialog(QWidget *parent, const QVector<BitTor
     m_ui->checkDisablePEX->setToolTip(privateTorrentsTooltip);
     m_ui->checkDisableLSD->setToolTip(privateTorrentsTooltip);
 
+    if (allSameSequential)
+        m_ui->checkSequential->setChecked(isFirstTorrentLSequentialEnabled);
+    else
+        m_ui->checkSequential->setCheckState(Qt::PartiallyChecked);
+
+    if (allSameFirstLastPieces)
+        m_ui->checkFirstLastPieces->setChecked(isFirstTorrentFirstLastPiecesEnabled);
+    else
+        m_ui->checkFirstLastPieces->setCheckState(Qt::PartiallyChecked);
+
     m_initialValues =
     {
+        m_ui->savePath->selectedPath(),
+        m_ui->comboCategory->currentText(),
         getRatio(),
         getSeedingTime(),
         m_ui->spinUploadLimit->value(),
         m_ui->spinDownloadLimit->value(),
+        m_ui->checkAutoTMM->checkState(),
         m_ui->checkDisableDHT->checkState(),
         m_ui->checkDisablePEX->checkState(),
-        m_ui->checkDisableLSD->checkState()
+        m_ui->checkDisableLSD->checkState(),
+        m_ui->checkSequential->checkState(),
+        m_ui->checkFirstLastPieces->checkState()
     };
 
+    // Needs to be called after the initial values struct is initialized
+    handleTMMChanged();
+
+    connect(m_ui->comboCategory, &QComboBox::currentIndexChanged, this, &TorrentOptionsDialog::handleCategoryChanged);
+    connect(m_ui->checkAutoTMM, &QCheckBox::clicked, this, &TorrentOptionsDialog::handleTMMChanged);
```

Comment:
preferably these should follow the order on UI, so this line should be atop of the previous line.